nipype/pipeline/engine/nodes.py

```diff
@@ -367,7 +368,22 @@ def run(self, updatehash=False):
             except:
                 os.remove(hashfile_unfinished)
                 raise
-            shutil.move(hashfile_unfinished, hashfile)
+
+            t = time()
+            timeout = float(self._config['execution']['job_finished_timeout'])
+            timed_out = True
+            while True:
+                try:
+                    shutil.move(hashfile_unfinished, hashfile)
+                    timed_out = False
+                    break
+                except Exception as e:
```

Comment:
So just to recap: the exception is thrown because **source file** `hashfile_unfinished` cannot be found, but the **target file** `hashfile` is where it should be as if the copying succeeded? If that the case we are likely dealing with two move operations happening at the same time. We should try to figure out why this was happening.